packages/firestore/test/unit/specs/spec_test_runner.ts

```diff
@@ -1180,7 +1180,13 @@ abstract class TestRunner {
         });
       }
 
-      expect(actual.view!.docChanges).to.deep.equal(expectedChanges);
+      expectedChanges = expectedChanges.sort((a, b) =>
```

Comment:
The sort() method sorts arrays in-place. It's probably not a good idea to do this, especially for `actual.view!.docChanges`. Instead, create a new array and sort it.

e.g. `const expectedChangesSorted = Array.from(expectedChanges).sort(...)`, and similarly for `actualChanges`.

This guarantees that you won't inadvertently affect some other part of the code by mutating these arrays.